docs/DeveloperGuide.md

```diff
@@ -0,0 +1,125 @@
+# RepoSense - Developer Guide
+Thank you for your interest in contributing to RepoSense!
+- [Setting up](#setting-up)
+  - [Prerequisites](#prerequisites)
+  - [Setting up the project in your computer using IntelliJ](#setting-up-the-project-in-your-computer-using-intellij)
+  - [Verifying the setup](#verifying-the-setup)
+  - [Configuring the coding style](#configuring-the-coding-style)
+- [Architecture](#architecture)
+  - [Parser](#parserconfigparser)
+  - [Git](#gitgitdownloader)
+  - [CommitsReporter](#commitsreporter)
+  - [AuthorshipReporter](#authorshipreporter)
+  - [ReportGenerator](#reportgeneratormain)
+  - [System](#system)
+  - [Model](#model)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+1. **JDK `1.8.0_60`** or later.
+1. **git `2.14`** or later on the command line.
+ > Type `git --version` on your OS terminal and ensure that you have the correct version of **git**.
+
+### Setting up the project in your computer using IntelliJ
+1. Fork this repo, and clone the fork to your computer.
+1. Open *IntelliJ* (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+1. Set up the correct *JDK* version for *Gradle*.
+    1. Click `Configure` > `Project Defaults` > `Project Structure`.
+    1. Click `New…​` and find the directory of the *JDK*.
+1. Click `Import Project`.
+1. Locate the `build.gradle` file and select it. Click `OK`.
+1. Ensure that the selected version of `Gradle JVM` matches our prerequisite.
+1. Click `OK` to accept the all the other default settings.
+
+### Verifying the setup
+1. Ensure that *Gradle* builds without error by running the command `gradlew clean build`, and ensure that it finishs with a `BUILD SUCCESSFUL` message.
+1. Run the tests to ensure they all pass by running the command `gradlew functional`, and ensure that it finishs with a `BUILD SUCCESSFUL` message
```

Comment:
missing `test`

Reply:
which part is missing `test`?

Reply:
because running `functional` alone might be hard to pinpoint wheres the error in case of a fail build.
adding `test` as well would be better.

Reply:
Hmmm, `build` already runs the tests though...
But since we mention `Run the tests` in this section, yep probably should add this in.

Reply:
@eugenepeh Done.